### PR TITLE
[CSL-2577] - Configure showing session modal conditionally after reaching results step

### PR DESCRIPTION
**Pull request by HHHindawy**

### CODE REVIEW

**Review by esezen — APPROVED** [review]
LGTM! Let's get another review from @mocca102

File: src/stories/Quiz/Component/QuizResults/QuizResults.stories.tsx [hunk 1]
```diff
@@ -37,8 +37,8 @@ export const QuizResultsPage: Story = {
     // 👇 Assert DOM structure
     const resultsTitle = canvas.getByText('Here are your results');
     const selectedOptions = document.querySelector('.cio-results-filter-options');
-    const resultsFilterChocolate = canvas.getByText('Chocolate');
-    const resultsFilterMedium = canvas.getByText('Medium');
+    const resultsFilterChocolate = canvas.getByText('Option 1');
+    const resultsFilterMedium = canvas.getByText('Option 2');
```
Comment (mocca102) [inline src/stories/Quiz/Component/QuizResults/QuizResults.stories.tsx:41]: Same here lets revert it back to "Medium"

File: src/hooks/useQuizEvents/index.ts [hunk 2]
```diff
@@ -23,6 +23,7 @@ const useQuizEvents: UseQuizEvents = (quizOptions, cioClient, quizState) => {
     dispatchLocalState,
     dispatchApiState,
     hasQuizStoredState,
+    isQuizCompleted,
```
Comment (mocca102) [inline src/hooks/useQuizEvents/index.ts:26]: If you notice here we're just passing `isQuizCompleted` to `useQuizEvents` and then return it as is. I think you did this following what is done for `hasQuizStoredState` But I think it's not necessary for it as well

Can we return those values from useQuizLocalState directly without passing them to useQuizEvents?

File: src/components/CioQuiz/index.tsx [hunk 3]
```diff
@@ -31,13 +31,19 @@ export default function CioQuiz(props: IQuizProps) {
   const [showSessionPrompt, setShowSessionPrompt] = useState(false);
   const { resultsPageOptions, sessionStateOptions } = props;
 
+  const skipToResults = isQuizCompleted() && !sessionStateOptions?.showSessionModalOnResults;
```
Comment (mocca102) [inline src/components/CioQuiz/index.tsx:34]: I think all we care about here is whether we will `skipToResults` or not. We don't care about exposing `isQuizCompleted` or `hasSessionStorageState` 

What do you think if we moved this logic inside the `useQuizLocalState` hook and only returned whether we should `skipToResults` or not?

**Review by mocca102 — CHANGES_REQUESTED** [review]

File: src/components/CioQuiz/index.tsx (hunk 3 above)
Comment (HHHindawy) [reply src/components/CioQuiz/index.tsx]: I like that 👍

**Review by mocca102 — APPROVED** [review]
LGTM!

**Pull request merged**